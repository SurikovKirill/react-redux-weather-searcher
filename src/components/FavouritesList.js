import React from 'react';
import '../containers/FavouriteContainer.css'
import FavouriteCityHeader from './FavouriteCityHeader';
import WeatherInfo from './WeatherInfo';
import Loader from './Loader';
import Error from './Error';

const FavouritesList = ({ cities, isLoading , errors, onRemove}) => (
    <div className='FavList'>
        {
            cities.map((city) => (
                <div key={city.name} className={'FavouriteCity'}>
                    <FavouriteCityHeader onRemove={onRemove} city={city}/>
                    {}
                    { (!errors.includes(city.name)) ? ((!isLoading.includes(city.name) ? <WeatherInfo weather={city}/> : <Loader/>)) : <Error/>}
                </div>
            ))
        }
    </div>
);

FavouritesList.defaultProps = {
    cities: [],
    isLoading: [],
    errors: [],
    onRemove: f => f
};


export default FavouritesList;