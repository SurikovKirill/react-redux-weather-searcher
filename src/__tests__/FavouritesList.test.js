import FavouritesList from '../components/FavouritesList';
import React from 'react';
import renderer from 'react-test-renderer';

let cities;
beforeEach(() => {
    cities = [
        {
            name: 'Moscow',
            img: 'i',
            temperature: 't',
            wind: 'w',
            cloudiness: 'c',
            pressure: 'p',
            humidity: 'h',
            location: 'l'
        },
        {
            name: 'Helsinki',
            img: 'i1',
            temperature: 't1',
            wind: 'w1',
            cloudiness: 'c1',
            pressure: 'p1',
            humidity: 'h1',
            location: 'l1'
        }
    ];
});
describe('FavoritesList component', () => {

    test('matches the snapshot when cities loaded', () => {

        const tree = renderer.create(
            <FavouritesList cities={cities} isLoading={[]} errors={[]}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });


    test('matches the snapshot when cities are loading', () => {

        const isLoading = ['Moscow'];

        const tree = renderer.create(
            <FavouritesList cities={cities} isLoading={isLoading} onRemove={f => f} errors={[]}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('matches the snapshot when on error', () => {

        const errors = ['Helsinki'];

        const tree = renderer.create(
            <FavouritesList cities={cities} isLoading={[]} onRemove={f => f} errors={errors}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});