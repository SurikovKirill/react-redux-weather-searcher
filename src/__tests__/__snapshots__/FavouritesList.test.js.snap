// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesList component matches the snapshot when cities are loading 1`] = `
<div
  className="FavList"
>
  <div
    className="FavouriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Moscow
      </h1>
      <img
        alt="Иконка погоды"
        src="i"
      />
      <h1>
        t
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="Loader"
    >
      <h1>
        Подождите, данные загружаются
      </h1>
      <div
        className="Spinner"
      />
    </div>
  </div>
  <div
    className="FavouriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Helsinki
      </h1>
      <img
        alt="Иконка погоды"
        src="i1"
      />
      <h1>
        t1
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="WeatherInfo"
    >
      <div
        className="Info"
      >
        <h3>
          Ветер
        </h3>
        <h4>
          w1
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Облачность
        </h3>
        <h4>
          c1
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Давление
        </h3>
        <h4>
          p1
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Влажность
        </h3>
        <h4>
          h1
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Координаты
        </h3>
        <h4>
          l1
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`FavoritesList component matches the snapshot when cities loaded 1`] = `
<div
  className="FavList"
>
  <div
    className="FavouriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Moscow
      </h1>
      <img
        alt="Иконка погоды"
        src="i"
      />
      <h1>
        t
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="WeatherInfo"
    >
      <div
        className="Info"
      >
        <h3>
          Ветер
        </h3>
        <h4>
          w
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Облачность
        </h3>
        <h4>
          c
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Давление
        </h3>
        <h4>
          p
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Влажность
        </h3>
        <h4>
          h
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Координаты
        </h3>
        <h4>
          l
        </h4>
      </div>
    </div>
  </div>
  <div
    className="FavouriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Helsinki
      </h1>
      <img
        alt="Иконка погоды"
        src="i1"
      />
      <h1>
        t1
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="WeatherInfo"
    >
      <div
        className="Info"
      >
        <h3>
          Ветер
        </h3>
        <h4>
          w1
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Облачность
        </h3>
        <h4>
          c1
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Давление
        </h3>
        <h4>
          p1
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Влажность
        </h3>
        <h4>
          h1
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Координаты
        </h3>
        <h4>
          l1
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`FavoritesList component matches the snapshot when on error 1`] = `
<div
  className="FavList"
>
  <div
    className="FavouriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Moscow
      </h1>
      <img
        alt="Иконка погоды"
        src="i"
      />
      <h1>
        t
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="WeatherInfo"
    >
      <div
        className="Info"
      >
        <h3>
          Ветер
        </h3>
        <h4>
          w
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Облачность
        </h3>
        <h4>
          c
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Давление
        </h3>
        <h4>
          p
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Влажность
        </h3>
        <h4>
          h
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Координаты
        </h3>
        <h4>
          l
        </h4>
      </div>
    </div>
  </div>
  <div
    className="FavouriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Helsinki
      </h1>
      <img
        alt="Иконка погоды"
        src="i1"
      />
      <h1>
        t1
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="Error"
    >
      <h1>
        Не удалось загрузить данные
      </h1>
    </div>
  </div>
</div>
`;
